worlds/blasphemous/Rules.py

```diff
@@ -1,4458 +1,1373 @@
+        return (
+            state.has("Tirana of the Celestial Bastion", self.player)
+            and self.total_fervour(state) >= 90
         )
-        or state.has_group("prayer", player)
-        or aubade(state, player)
-        or tirana(state, player)
-    )
-
-
-def can_break_tirana(state: CollectionState, logic: int, player: int) -> bool:
-    return tirana(state, player) if logic >= 2 else False
-
-
-def can_dive_laser(state: CollectionState, logic: int, player: int) -> bool:
-    return state.has("Dive Skill", player, 3) if logic >= 2 else False
-
-
-def can_walk_on_root(state: CollectionState, player: int) -> bool:
-    return state.has("Three Gnarled Tongues", player)
-
-
-def can_climb_on_root(state: CollectionState, player: int) -> bool:
-    return state.has_all({"Three Gnarled Tongues", "Wall Climb Ability"}, player)
-
-
-def can_survive_poison(state: CollectionState, logic: int, player: int, number: int) -> bool:
-    if number == 1:
-        if logic >= 2:
-            return True
-        elif logic == 1:
-            return state.has_any({"Silvered Lung of Dolphos", "Tiento to your Thorned Hairs"}, player)
-        elif logic == 0:
-            return state.has("Silvered Lung of Dolphos", player)
-    elif number == 2:
-        if logic >= 1:
-            return state.has_any({"Silvered Lung of Dolphos", "Tiento to your Thorned Hairs"}, player)
-        else:
-            return state.has("Silvered Lung of Dolphos", player)
-    elif number == 3:
-        if logic >= 2 and total_fervour(state, player) >= 120:
-            return state.has_any({"Silvered Lung of Dolphos", "Tiento to your Thorned Hairs"}, player)
-        else:
-            return state.has("Silvered Lung of Dolphos", player)
 
+    def ruby(self, state: CollectionState) -> bool:
+        return state.has("Cloistered Ruby", self.player)
+    
+    def tiento(self, state: CollectionState) -> bool:
+        return state.has("Tiento to my Sister", self.player)
+    
+    def any_small_prayer(self, state: CollectionState) -> bool:
+        return (
+            self.debla(state)
+            or self.lorquiana(state)
+            or self.zarabanda(state)
+            or self.taranto(state)
+            or self.verdiales(state)
+            or self.cante(state)
+            or self.cantina(state)
+            or self.tiento(state)
+        )
+    
+    def pillar(self, state: CollectionState) -> bool:
+        return (
+            self.debla(state)
+            or self.taranto(state)
+            or self.ruby(state)
+        )
+    
+    def can_use_any_prayer(self, state: CollectionState) -> bool:
+        return (
+            self.any_small_prayer(state)
+            or self.tirana(state)
+            or self.aubade(state)
+        )
 
-def can_enemy_bounce(logic: int, enemy: int) -> bool: # TODO
-    return enemy_skips_allowed(logic, enemy)
+    # Stats
+    def total_fervour(self, state: CollectionState) -> int:
```

Comment:
Should this cap how much each contributes? Though I don't know how the client handles something like 10 Beads of Blue Wax